### PR TITLE
fix: optimize isNullAt

### PR DESCRIPTION
**Pull request by kazuyukitanimura**
## Which issue does this PR close?

Part of #679 and #670

## Rationale for this change

This PR improves the isNullAt performance

## What changes are included in this PR?

The new `isNullAt ` method uses the OR operator instead of the if clause

## How are these changes tested?

Existing tests

### CODE REVIEW

**Comment by kazuyukitanimura** [comment]
SF=100
### Before
Screenshot 2024-07-26 at 4 45 07 PM

### After
Screenshot 2024-07-26 at 4 45 28 PM

File: common/src/main/java/org/apache/comet/vector/CometPlainVector.java [hunk 1]
```diff
@@ -153,11 +153,7 @@ public CDataDictionaryProvider getDictionaryProvider() {
 
   @Override
   public boolean isNullAt(int rowId) {
-    if (this.valueBufferAddress == -1) {
-      return true;
-    } else {
-      return super.isNullAt(rowId);
-    }
+    return this.valueBufferAddress == -1 || super.isNullAt(rowId);
```
Comment (parthchandra) [inline common/src/main/java/org/apache/comet/vector/CometPlainVector.java:156]: I would expect this to have identical performance to the original (i.e. this should result in identical byte code).

Comment (kazuyukitanimura) [reply common/src/main/java/org/apache/comet/vector/CometPlainVector.java:156]: ### Before
```
  public boolean isNullAt(int);
    Code:
       0: aload_0
       1: getfield      #45                 // Field valueBufferAddress:J
       4: ldc2_w        #42                 // long -1l
       7: lcmp
       8: ifne          13
      11: iconst_1
      12: ireturn
      13: aload_0
      14: iload_1
      15: invokespecial #195                // Method org/apache/comet/vector/CometDecodedVector.isNullAt:(I)Z
      18: ireturn
```

### After
```
  public boolean isNullAt(int);
    Code:
       0: aload_0
       1: getfield      #45                 // Field valueBufferAddress:J
       4: ldc2_w        #42                 // long -1l
       7: lcmp
       8: ifeq          19
      11: aload_0
      12: iload_1
      13: invokespecial #195                // Method org/apache/comet/vector/CometDecodedVector.isNullAt:(I)Z
      16: ifeq          23
      19: iconst_1
      20: goto          24
      23: iconst_0
      24: ireturn
```

So basically jump order is different and seems that is causing the difference at least on my machine...

Comment (parthchandra) [reply common/src/main/java/org/apache/comet/vector/CometPlainVector.java:156]: interesting find

**Review by parthchandra — APPROVED** [review]

**Comment by kazuyukitanimura** [comment]
cc @andygrove @comphead @huaxingao @viirya

**Review by viirya — APPROVED** [review]

File: common/src/main/java/org/apache/comet/vector/CometPlainVector.java [hunk 2]
```diff
@@ -153,11 +153,7 @@ public CDataDictionaryProvider getDictionaryProvider() {
 
   @Override
   public boolean isNullAt(int rowId) {
-    if (this.valueBufferAddress == -1) {
```
Comment (viirya) [inline common/src/main/java/org/apache/comet/vector/CometPlainVector.java:156]: I'm surprised that Java compiler cannot optimize this.

**Review by andygrove — APPROVED** [review]

**Comment by kazuyukitanimura** [comment]
Thanks

**Pull request merged**